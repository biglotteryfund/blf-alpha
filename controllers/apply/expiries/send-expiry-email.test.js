/* eslint-env jest */
'use strict';
const nodemailer = require('nodemailer');
const moment = require('moment');

const { sendExpiryEmail } = require('./send-expiry-email');

const mockTransport = nodemailer.createTransport({
    jsonTransport: true,
});

test('Apply for funding under £10,000 expiry email', async function () {
    const info = await sendExpiryEmail(
        {
            emailType: 'UNDER10K_ONE_MONTH',
            unsubscribeToken: 'MOCK_TOKEN',
            formId: 'awards-for-all',
            applicationId: 'MOCK_APPLICATION_ID',
            applicationData: {
                projectCountry: 'england',
                projectName: 'Example project name',
            },
            expiresAt: moment('2050-06-01 12:00').toISOString(),
            sendTo: 'example@example.com',
        },
        mockTransport
    );

    const infoMessage = JSON.parse(info.message);

    expect(infoMessage.subject).toBe(
        'You have one month to finish your application'
    );
    expect(infoMessage.text).toMatchSnapshot();
});

test('Apply for funding under £10,000 welsh expiry email', async function () {
    const info = await sendExpiryEmail(
        {
            emailType: 'UNDER10K_ONE_MONTH',
            unsubscribeToken: 'MOCK_TOKEN',
            formId: 'awards-for-all',
            applicationId: 'MOCK_APPLICATION_ID',
            applicationData: {
                projectCountry: 'wales',
                projectName: 'Example project name',
            },
            expiresAt: moment('2050-06-01 12:00').toISOString(),
            sendTo: 'example@example.com',
        },
        mockTransport
    );

    const infoMessage = JSON.parse(info.message);

    expect(infoMessage.subject).toBe(
        'You have one month to finish your application / Mae gennych fis i orffen eich cais'
    );
    expect(infoMessage.text).toMatchSnapshot();
});

test('Your funding proposal expiry email', async function () {
    const info = await sendExpiryEmail(
        {
            emailType: 'STANDARD_ONE_MONTH',
            unsubscribeToken: 'MOCK_TOKEN',
            formId: 'standard-enquiry',
            applicationId: 'MOCK_APPLICATION_ID',
            applicationData: {
                projectCountries: ['england'],
                projectName: 'Example project name',
            },
            expiresAt: moment('2050-06-01 12:00').toISOString(),
            sendTo: 'example@example.com',
        },
        mockTransport
    );

    const infoMessage = JSON.parse(info.message);

    expect(infoMessage.subject).toBe(
        'You have one month to finish your funding proposal'
    );
    expect(infoMessage.text).toMatchSnapshot();
});

test('Your funding proposal welsh expiry email ', async function () {
    const info = await sendExpiryEmail(
        {
            emailType: 'STANDARD_ONE_MONTH',
            unsubscribeToken: 'MOCK_TOKEN',
            formId: 'standard-enquiry',
            applicationId: 'MOCK_APPLICATION_ID',
            applicationData: {
                projectCountries: ['wales'],
                projectName: 'Example project name',
            },
            expiresAt: moment('2050-06-01 12:00').toISOString(),
            sendTo: 'example@example.com',
        },
        mockTransport
    );

    const infoMessage = JSON.parse(info.message);

    expect(infoMessage.subject).toBe(
        'You have one month to finish your funding proposal / Mae gennych fis ar ôl i orffen eich cynnig'
    );
    expect(infoMessage.text).toMatchSnapshot();
});
