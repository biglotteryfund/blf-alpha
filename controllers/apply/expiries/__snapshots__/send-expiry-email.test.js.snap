// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply for funding under £10,000 expiry email 1`] = `
" Hello,

 Your application for under £10,000 isn’t finished yet
We noticed you started applying for Example project name, but haven’t finished yet. 

 You have until 1 June, 2050 12:00pm to finish your application 
If you don’t complete it before that time, we’ll delete it, so we can keep your data safe. 

 How to finish your application
Log in to your account [https://www.tnlcommunityfund.org.uk/apply/under-10k/edit/MOCK_APPLICATION_ID?s=expiryEmail]and answer the
rest of the questions. 

 If you don’t want more reminders from us
You can tell us to stop sending you these reminder emails.
[https://www.tnlcommunityfund.org.uk/apply/emails/unsubscribe?token=MOCK_TOKEN] 

 If you have any questions at all
You can call 028 9568 0143. Or email general.enquiries@tnlcommunityfund.org.uk. We’ll be happy to help you. 

 Best wishes,
The National Lottery Community Fund"
`;

exports[`Apply for funding under £10,000 welsh expiry email 1`] = `
" Bilingual message – please scroll down for Welsh.
Neges ddwyieithog – y Gymraeg i’w weld isod. 

Hello,

 Your application for under £10,000 isn’t finished yet
We noticed you started applying for Example project name, but haven’t finished yet. 

 You have until 1 June, 2050 12:00pm to finish your application 
If you don’t complete it before that time, we’ll delete it, so we can keep your data safe. 

 How to finish your application
Log in to your account [https://www.tnlcommunityfund.org.uk/apply/under-10k/edit/MOCK_APPLICATION_ID?s=expiryEmail]and answer the
rest of the questions. 

 If you don’t want more reminders from us
You can tell us to stop sending you these reminder emails.
[https://www.tnlcommunityfund.org.uk/apply/emails/unsubscribe?token=MOCK_TOKEN] 

 If you have any questions at all
You can call 029 2168 0214. Or email wales@tnlcommunityfund.org.uk. We’ll be happy to help you. 

 Best wishes,
The National Lottery Community Fund 


----------------------------------------------------------------------------------------------------------------------------------

Helo,

 Nid yw'ch cais am lai na £10,000 wedi'i orffen eto
Fe sylweddolom eich bod wedi dechrau ymgeisio am Example project name, ond heb orffen eich cais. 

 Mae gennych tan 1 Mehefin, 2050 12:00pm i orffen eich cais 
Os nad ydych yn ei gwblhau yn yr amser yna, byddwn yn ei ddileu, er mwyn i ni gadw eich data’n ddiogel. 

 Sut i orffen eich cais Mewngofnodi i’ch cyfrif
[https://www.tnlcommunityfund.org.uk/welsh//apply/under-10k/edit/MOCK_APPLICATION_ID?s=expiryEmail]ac ateb gweddill y cwestiynau. 

 Os nad ydych eisiau rhagor o nodiadau atgoffa gennym
Gallwch ddweud wrthym i stopio anfon yr e-byst atgoffa hyn atoch
[https://www.tnlcommunityfund.org.uk/welsh/apply/emails/unsubscribe?token=MOCK_TOKEN]. 

 Os oes gennych unrhyw gwestiynau
Gallwch ffonio 029 2168 0214. Neu e-bostio wales@tnlcommunityfund.org.uk. Byddwn yn hapus i’ch helpu. 

 Cofion gorau,
Cronfa Gymunedol y Loteri Genedlaethol"
`;

exports[`Your funding proposal expiry email 1`] = `
" Hello,

 Your funding proposal isn’t finished yet
We noticed you started telling us about Example project name. But you haven’t finished yet. 

 You have until 1 June, 2050 12:00pm to finish your funding proposal. 
If you don’t complete it before that time, we’ll delete it - so that we can keep your data safe. 

 How to finish your funding proposal
Log in to your account [https://www.tnlcommunityfund.org.uk/apply/your-funding-proposal/edit/MOCK_APPLICATION_ID?s=expiryEmail]and
answer the rest of the questions. 

 If you don’t want more reminders from us
You can tell us to stop sending you these reminder emails.
[https://www.tnlcommunityfund.org.uk/apply/emails/unsubscribe?token=MOCK_TOKEN] 

 If you have any questions at all
You can call 028 9568 0143. Or email general.enquiries@tnlcommunityfund.org.uk. We’ll be happy to help you. 

Best wishes,
The National Lottery Community Fund"
`;

exports[`Your funding proposal welsh expiry email  1`] = `
" Bilingual message – please scroll down for Welsh.
Neges ddwyieithog – y Gymraeg i’w weld isod. 

Hello,

 Your funding proposal isn’t finished yet
We noticed you started telling us about Example project name. But you haven’t finished yet. 

 You have until 1 June, 2050 12:00pm to finish your funding proposal. 
If you don’t complete it before that time, we’ll delete it - so that we can keep your data safe. 

 How to finish your funding proposal
Log in to your account [https://www.tnlcommunityfund.org.uk/apply/your-funding-proposal/edit/MOCK_APPLICATION_ID?s=expiryEmail]and
answer the rest of the questions. 

 If you don’t want more reminders from us
You can tell us to stop sending you these reminder emails.
[https://www.tnlcommunityfund.org.uk/apply/emails/unsubscribe?token=MOCK_TOKEN] 

 If you have any questions at all
You can call 029 2168 0214. Or email wales@tnlcommunityfund.org.uk. We’ll be happy to help you. 

Best wishes,
The National Lottery Community Fund 

Hello,

 Nid yw eich cynnig ariannu wedi gorffen eto
Fe sylweddolom eich bod wedi dechrau dweud wrthym am Example project name. Ond nid ydych wedi gorffen eto. 

 Mae gennych tan 1 June, 2050 12:00pm i orffen eich cynnig ariannu. 
Os nad ydych yn ei gwblhau yn yr amser yna, byddwn yn ei ddileu - er mwyn i ni allu cadw eich data yn ddiogel. 

 Sut i orffen eich cynnig ariannu
Mewngofnodwch i’ch cyfrif [https://www.tnlcommunityfund.org.uk/apply/your-funding-proposal/edit/MOCK_APPLICATION_ID?s=expiryEmail]
ac ateb gweddill y cwestiynau. 

 Os nad ydych eisiau mwy o nodiadau atgoffa gennym
Gallwch ddweud wrthym i stopio anfon yr e-byst atgoffa hyn
[https://www.tnlcommunityfund.org.uk/apply/emails/unsubscribe?token=MOCK_TOKEN] 

 Os oes gennych unrhyw gwestiynau
Gallwch ffonio 029 2168 0214. Neu e-bostio wales@tnlcommunityfund.org.uk. Byddwn yn hapus i’ch helpu. 

Cofion gorau,
Cronfa Gymunedol y Loteri Genedlaethol"
`;
